### ISSUE TITLE
Add support for B-D buttons

### ISSUE THREAD

**Issue by vmikhav** [Catuna/AndroidMediaControlsWindows#2]
- [ ] Experimentally determine the amplitude of the signal when you press buttons B-D.
- [ ] Add the definition of the receipt of a signal of such amplitude on the audio input.
- [ ] Add event handlers for additional buttons.

![Headset circuit](http://www.roligheten.no/images/headset-circuit2.png)

**Comment by ghost** [comment]
Yup, was looking up how to have button support on pc/laptop. It worked on my bluetooth headset (although I can't do it when task manager is on for some reason), but it didn't work on my wired headset. Button is still recorded as sound, I think this has something to do with it.

**Comment by PaperStrike** [comment]
This is not possible by hardware. Although resistors between the buttons are different, the effect showed on waveform will be almost the same - pressing buttons cause unusually huge voltage difference compared to normal voice while our pc is designed to process normal voice only.

**Comment by roligheten** [comment]
Pretty sure it is indeed not possible to detect the different buttons using only the signal amplitude, but since I am not an electronics guy I am keeping this open until someone can give a good explanation why it is not. 🙂

**Comment by PaperStrike** [comment]
Somehow I found a **distinguishable** difference between normal voice and each buttons by the signal amplitudes today. 😅
What I said yesterday seems wrong.

By comparing the two channel of the mic, and draw a x-y graph: (Software I used here: [Soundcard Oscilloscope](https://www.zeitnitz.eu/scope_en))

- the normal voice's graph is **always** like a `y = x` curve: (I have tried my best to make it bold, but it **never** reach -0.045 on the left.)

    ![Button0](https://user-images.githubusercontent.com/22674396/107115380-42887b80-68a7-11eb-8651-ea599887e387.png)

While, each button's graph adds interference to the `y = x` with different amplitude. you may pay attention to the **y-axis**, especially values range **[-0.045, -0.02]**.

1. Button A's graph:

    ![ButtonA](https://user-images.githubusercontent.com/22674396/107115426-a0b55e80-68a7-11eb-8410-149f866bd5b5.png)

2. Button B's graph:

    ![ButtonB](https://user-images.githubusercontent.com/22674396/107115434-b0cd3e00-68a7-11eb-977d-939ee350dcfd.png)

3. Button C's graph: (always above -0.045 on the left. I stress this to prove it is distinguishable from the normal voice.)

    ![ButtonC](https://user-images.githubusercontent.com/22674396/107115442-bd519680-68a7-11eb-82c9-2efe19e78955.png)

My headset doesn't have a Button D. But I guess its graph is between Button A and B's, since its impedance is between theirs.

I'm not an expert in programming, hope these information does some help.

**Comment by PaperStrike** [comment]
Another finding by comparing the two channel of the mic:
**now we can easily tell a long press!** 🎉

If we keep pressing a button, its graph would be like a ball:

![ButtonPressing](https://user-images.githubusercontent.com/22674396/107115513-220cf100-68a8-11eb-8d84-b98cafce47d0.png)


We may check the the two channel continuously, counting the result of `channel2 - channel1`, and once the result becomes larger than a threshold, we can tell the press has start. when the absolute value of the result reach below the threshold for some time, the press has end. The largest result can tell which button is pressed, the duration can tell if it's a long press!

**Comment by roligheten** [comment]
@PaperFlu That's really cool! I haven't had time to look into this in detail but I will try to investigate it myself when I have time.
